firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/ApkInstaller.java

```diff
@@ -111,6 +111,7 @@ private void startInstallActivity(String path, Activity currentActivity) {
     }
     Intent intent = new Intent(currentActivity, InstallActivity.class);
     intent.putExtra("INSTALL_PATH", path);
+    intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
```

Comment:
Why add this flag?

Reply:
Good catch! Leftover from trying to use application context. When using application Context it needs to be a new task but since that was taken out this should also be taken out.

Reply:
Oh interesting. Why does it need to a be a new task when using application context?

Reply:
I ran into the issue but this post sums up the issue and the reason for the fix. https://stackoverflow.com/questions/3918517/calling-startactivity-from-outside-of-an-activity-context

Reply:
Hmm, ok the fact that you saw that error is making me less confident in my suggestion to use intents + activities. It feels weird to have to start a new task every time.

Reply:
We do this in the AAB updater https://github.com/firebase/firebase-android-sdk/blob/712e5b88c75e6ad8838d21c904a789f37bcfa0a7/firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/AabUpdater.java#L160 With the application context we aren't guaranteed an activity so the flag makes sense to me.

Reply:
I think it makes more sense in the AAB updater since we are launching an activity in a different app, Play.

In the case I was proposing, where we would launch a new intent/activity in order to simply show a dialog in the app, we wouldn't want an additional entry in the task backstack for that. It should ideally just continue in the same task.

Reply:
This error is especially interesting, since we should be starting the install activity as a new task anyway according to the manifest I think https://github.com/firebase/firebase-android-sdk/blob/master/firebase-app-distribution/src/main/AndroidManifest.xml#L34-L35

Reply:
From the docs:
`singleTask`
The system creates a new task and instantiates the activity at the root of the new task.